lib/percona_migrator/lhm/fake/adapter.rb

```diff
@@ -23,14 +23,18 @@ def initialize(migration, table_name)
         # @param column_name [String, Symbol]
         # @param definition [String]
         def add_column(column_name, definition)
-          column = column(column_name, definition)
-
-          migration.add_column(
-            table_name,
-            column_name,
-            column.type,
-            column.to_hash
-          )
+          if definition.is_a?(Symbol)
+            migration.add_column(table_name, column_name, definition)
+          else
+            column = column(column_name, definition)
+
+            migration.add_column(
+              table_name,
+              column_name,
+              column.type,
+              column.to_hash
+            )
+          end
```

Comment:
Yes, the different number of attributes involves something like that. However, it'd be simpler to pass an empty options hash (last argument) if possible. In any case, I'll try to dry that if.